### PR TITLE
sql: stop observing the CommitTimestamp in TRUNCATE

### LINKED ISSUE #41566: teamcity: failed test: TestChangefeedTruncateRenameDrop
The following tests appear to have failed on release-19.2 (testrace): TestChangefeedTruncateRenameDrop, TestChangefeedTruncateRenameDrop/sinkless, TestChangefeedTruncateRenameDrop/enterprise

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestChangefeedTruncateRenameDrop).

[#1537922](https://teamcity.cockroachdb.com/viewLog.html?buildId=1537922):

```
TestChangefeedTruncateRenameDrop/enterprise
... minutes]:
database/sql.(*DB).connectionOpener(0xc00db42a80, 0x6a76780, 0xc00accc080)
	/usr/local/go/src/database/sql/sql.go:1000 +0x140
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:670 +0x2b4

goroutine 63441 [chan receive]:
github.com/cockroachdb/cockroach/pkg/storage/closedts/provider.(*Provider).Notify.func1(0x6a76840, 0xc001a40510)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/closedts/provider/provider.go:217 +0x1ae
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc006cd27e0, 0xc00abe86e0, 0xc0092790a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 63696 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
sync.runtime_notifyListWait(0xc008a37890, 0xc000025ea0)
	/usr/local/go/src/runtime/sema.go:510 +0xf9
sync.(*Cond).Wait(0xc008a37880)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc0010314d0, 0x6a76840, 0xc001a40c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:192 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x6a76840, 0xc001a40c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:161 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc0068ccbf0, 0xc00abe86e0, 0xc0068ccbe0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 65120 [chan receive, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).RangeFeed(0xc00465a900, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_rangefeed.go:229 +0x90d
github.com/cockroachdb/cockroach/pkg/storage.(*Store).RangeFeed(0xc00147bc00, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2213 +0x4c6
github.com/cockroachdb/cockroach/pkg/storage.(*Stores).RangeFeed(0xc007531550, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x5ab6980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/stores.go:206 +0x24b
github.com/cockroachdb/cockroach/pkg/server.(*Node).RangeFeed(0xc0089d6a00, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x6a76780, 0xc002a96300)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:1033 +0x7e
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed.func1(0xc00a2fb870, 0x6a30100, 0xc0089d6a00, 0xc00c00b900, 0x6a76780, 0xc002a96480, 0xc00c356240, 0xc00c3562a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:610 +0xcc
created by github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:608 +0x12e

goroutine 63745 [select, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/server.periodicallyPersistHLCUpperBound(0xc00129f4f0, 0xc003923340, 0xc0068cd900, 0x5d09da8, 0xc008a8a480, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1030 +0x1f8
github.com/cockroachdb/cockroach/pkg/server.(*Server).startPersistingHLCUpperBound.func2(0x6a76840, 0xc0018b5f80)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1104 +0xde
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc0068cd910, 0xc00abe86e0, 0xc002910690)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4
I191014 14:23:02.501369 1 rand.go:85  Random seed: -3946147195552034504



TestChangefeedTruncateRenameDrop
--- FAIL: testrace/TestChangefeedTruncateRenameDrop (0.000s)
Test ended in panic.




TestChangefeedTruncateRenameDrop/sinkless
...is range at key /Table/60 [r30] (zone config)
I191014 14:26:51.237618 62064 sql/event_log.go:130  [n1,client=127.0.0.1:43882,user=root] Event: "create_table", target: 60, info: {TableName:d.public.drop Statement:CREATE TABLE drop (a INT8 PRIMARY KEY) User:root}
I191014 14:26:51.954272 62064 sql/sqlbase/structured.go:1516  [n1,client=127.0.0.1:43882,user=root] publish: descID=60 (drop) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I191014 14:26:52.156068 62064 sql/event_log.go:130  [n1,client=127.0.0.1:43882,user=root] Event: "drop_table", target: 60, info: {TableName:d.public.drop Statement:DROP TABLE drop User:root CascadeDroppedViews:[]}
I191014 14:26:52.225290 62064 sql/sqlbase/structured.go:1516  [n1,client=127.0.0.1:43882,user=root,scExec] publish: descID=60 (drop) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I191014 14:26:52.749493 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] storage.IntentResolver: cleanup txn records
13     gossip update (merge)
10     gossip update (split)
1      [async] transport racer
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.749900 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] storage.IntentResolver: cleanup txn records
13     gossip update (merge)
10     gossip update (split)
1      [async] transport racer
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
W191014 14:26:52.750060 63439 storage/intentresolver/intent_resolver.go:818  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
I191014 14:26:52.755778 61528 storage/queue.go:520  [n1,s1] rate limited in gossip update (merge): node unavailable; try another peer
I191014 14:26:52.772680 61388 kv/transport_race.go:108  transport race promotion: ran 18 iterations on up to 1120 requests
I191014 14:26:52.773013 61341 util/stop/stopper.go:542  quiescing; tasks left:
5      gossip update (split)
3      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774161 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
3      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774449 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
2      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774709 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
2      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774988 61341 util/stop/stopper.go:542  quiescing; tasks left:
3      gossip update (split)
2      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775233 61341 util/stop/stopper.go:542  quiescing; tasks left:
3      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775539 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775849 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.776037 61528 storage/queue.go:520  [n1,s1] rate limited in gossip update (split): node unavailable; try another peer
I191014 14:26:52.776500 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.777106 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber




```

Please assign, take a look and update the issue accordingly.

### PR DESCRIPTION
**Pull request by ajwerner**
In #40581 we stopped observing the commit timestamp to write it into table
descriptors. In this change I overlooked (rather forgot) about this additional
place in the code where we observed the commit timestamp. As far as I can tell
we don't read this field anywhere ever. Furthermore we know that the the table
descriptor in question to which we are referring must be alive and equal to
the provided value at the timestamp at which it was read due to serializability.
In short, this minor change continues to populate the field with a sensible
value and will permit TRUNCATE to be pushed.

Fixes #41566.

Release note (bug fix): Long running transactions which attempt to TRUNCATE
can now be pushed and will commit in cases where they previously could fail
or retry forever.

### CODE REVIEW

**Review by andreimatei — COMMENTED** [review]
LGTM

Nit: I wouldn't talk about observing commit timestamps in the release note. And also I don't think phrasing like "this change" is appropriate in release notes - there's no "this" in that context.

Lucy, can we get an explanation of the Time field in the `ReplacementOf` struct please? It has no comment... Let's make it wall time instead of an hlc.Timestamp if it doesn't need to be HLC.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42650#-:-LuDc8vn8bdtr6yioMHL:b-yusiw6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @lucy-zhang)

**Comment by ajwerner** [comment]
I added some commentary and updated the release note.

**Comment by ajwerner** [comment]
@lucy-zhang friendly ping

**Review by thoszhang — APPROVED** [review]
> Reviewed 2 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42650)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @lucy-zhang)

**Comment by ajwerner** [comment]
TFTR!

bors r+
